### PR TITLE
Add Qwen3 reranker classify pooling

### PR DESCRIPTION
**Pull request by CAICAIIs**
## Summary

Adds a narrow Metal V1 text pooling follow-up for original Qwen3 reranker `classify` support.

This PR supports only the Qwen3 reranker contract:

- `Qwen3ForSequenceClassification`
- `classifier_from_token=["no", "yes"]`
- `is_original_qwen3_reranker=True`
- tied `model.model.embed_tokens.as_linear`
- paged pooling path

The score is computed from the final prompt-token hidden state using `yes_logit - no_logit`, with optional `logit_mean` / `logit_sigma` calibration and `use_activation` handling.

## Scope

This intentionally does not add generic classification or reranking support. Generic classification heads, generic reranking models, late interaction, token-level pooling, multimodal pooling, direct embedding tensors, and non-paged pooling remain unsupported.

## Changes

- Adds Qwen3 reranker `classify` pooling helpers in `pooling.py`
- Keeps `model_runner.py` as a thin router/delegator
- Adds contract tests for task advertisement, output order, raw/activated scores, calibration, and fail-fast behavior
- Updates text pooling docs and supported model evidence

## Validation

- `git diff --check`
- `.venv-vllm-metal/bin/python -m pytest tests/test_v1_pooling.py -q`
  - `32 passed`
- `.venv-vllm-metal/bin/python -m pytest tests/test_v1_model_runner_generate.py tests/test_v1_worker.py tests/test_paged_attention.py tests/test_sampling.py tests/test_grammar_bitmask.py -q`
  - `119 passed`
- `scripts/lint.sh`
  - shellcheck, ruff, ruff format check, and mypy passed

Real Qwen3 reranker smoke:

- Offline `LLM.score` on `mku64/Qwen3-Reranker-0.6B-mlx-8Bit`
  - revision `ba80418a47fa1c4368a6c2287b0e449904063576`
  - `reranker-offline-score-ok 2 [0.973079, 0.931326]`
- Server `/score`
  - `/health` returned HTTP 200
  - `/score` returned HTTP 200 with two scalar scores

Refs #361

### CODE REVIEW

File: vllm_metal/v1/pooling.py [hunk 1]
```diff
@@ -109,12 +121,30 @@ def _is_decoder_embedding_config(model_config: Any) -> bool:
...
+    )
+
+
 def sequence_model(model: Any) -> Any | None:
     """Return the MLX transformer body when it is available."""
     inner = getattr(model, "model", None)
     return inner if callable(inner) else None
 
 
+def _classifier_logits_fn(model: Any) -> Any | None:
```
Comment (WindChimeRan) [inline vllm_metal/v1/pooling.py]: The PR always computes them via `embed_tokens.as_linear`,  which is the correct output head only when tied. On an untied model (like Qwen3-Reranker-8B), the real head is the separate lm_head, so projecting through embed_tokens gives the wrong logits.

Comment (CAICAIIs) [reply vllm_metal/v1/pooling.py]: Sorry, you were right. I had incorrectly assumed `embed_tokens.as_linear` was always the correct projection head.

I’ve updated the PR so untied checkpoints use the separate `lm_head`, while tied checkpoints use `embed_tokens.as_linear`. If the tie state is unknown, the model is no longer advertised as supported. I also added targeted tests covering the untied `lm_head` path and unsupported fallback cases.

Thanks for catching this.

**Review by WindChimeRan — APPROVED** [review]
Please don't squash mulitple commits into one commit next time. 

keeping commit history will make review easier.

**Comment by CAICAIIs** [comment]
> Please don't squash mulitple commits into one commit next time.
> 
> keeping commit history will make review easier.

oh, sorry, got it!
